formats/json/commonMain/src/kotlinx/serialization/json/internal/StreamingJsonDecoder.kt

```diff
@@ -87,6 +87,9 @@ internal open class StreamingJsonDecoder(
             return result
 
         } catch (e: MissingFieldException) {
+            // Add "at path" if and only if we've just caught an exception and it hasn't been augmented yet
+            if (e.message!!.contains("at path")) throw e
+            // NB: we could've use some additional flag marker or augment the stacktrace, but it seemed to be as too much of a burden
             throw MissingFieldException(e.missingFields, e.message + " at path: " + lexer.path.getPath(), e)
```

Comment:
Can we inherit from class `MissingFieldException` or will it break backward compatibility?

Reply:
We can, but it's a whole new class for the sake of very minor thing, I believe this one is more than enough